scripts/cocoapods/__tests__/utils-test.rb

```diff
@@ -382,6 +382,44 @@ def test_fixReactBridgingHeaderSearchPaths_correctlySetsTheHeaderSearchPathsForA
         end
     end
 
+    # ===================================== #
+    # Test - Apply Xcode14 React-Core patch #
+    # ===================================== #
+
+    def test_turnOffResourceBundleReactCore_correctlyAppliesPatch
+        # Arrange
+        react_core_debug_config = prepare_Code_Signing_YES_build_configuration("Debug")
+        react_core_release_config = prepare_Code_Signing_YES_build_configuration("Release")
+
+        user_project_mock = UserProjectMock.new("a/path", [
+            prepare_config("Debug"),
+            prepare_config("Release"),
+        ])
+
+        pods_projects_mock = PodsProjectMock.new([],
+            {
+            "hermes-engine" => {},
+            "React-Core" => [ react_core_debug_config, react_core_release_config ],
```

Comment:
I would aslo add another target which is not React-Core and which has the BuildConfigurationMock with the `CODE_SIGNING_ALLOWED` set to YES. And we assert that it stays to YES. In this way we test that we are actually only changing the React-Core pod.